Formula/aacgain.rb

```diff
@@ -32,4 +32,9 @@ def install
     cp test_fixtures("test.m4a"), "test.m4a"
     system bin/"aacgain", "test.m4a"
   end
+
+  livecheck do
+    url "https://aacgain.altosdesign.com/alvarez/"
+    regex(/href=.+?aacgain-(\d+(?:\.\d+)+)\.t/)
```

Comment:
Could the `alvarez` be part of the `regex` and this use the `homepage` somehow?

Reply:
Livecheck uses the `page_match` strategy for this URL, so it's using the regex to match versions within the HTML of the specified page. The content of the homepage and `/alvaraez/` page are completely different, so no.

Additionally, we generally try to do the check as close to the stable archive as we can. In this case, the current stable archive is https://aacgain.altosdesign.com/alvarez/aacgain-1.8.tar.bz2 and https://aacgain.altosdesign.com/alvarez/ is an index page listing archive files.

It's easy to reliably identify versions on an index page like this, whereas the homepage text is potentially unreliable. Basically, the regex you would need to try to match versions on the homepage text would be loose (potentially matching more than it should) and if they homepage text notably changes (not uncommon) the check would break and need to be updated.

For what it's worth, I went through the existing livecheckables yesterday and fixed any that were broken, so the existing livecheckables are currently working properly (outside of the ones where upstream is unreliable and fails on some runs)

Reply:
> Livecheck uses the `page_match` strategy for this URL

Stupid question: how do you know that from this DSL?



> The content of the homepage and `/alvaraez/` page are completely different, so no.

👍

Reply:
> how do you know that from this DSL?

The livecheckable doesn't explicitly mention the strategy at this point (the optional "using" feature will allow you to make it explicit, once it's implemented), so it's basically just that I know how the heuristic works and I check the output of `brew livecheck --debug aacgain`.

The output varies depending on the strategy but you're generally looking for something after the URL that hints at the strategy. `page_match` is pretty straightforward to identify:

```
Trying with url https://aacgain.altosdesign.com/alvarez/
Using page_match("https://aacgain.altosdesign.com/alvarez/", "(?-mix:href=.+?aacgain-(\d+(?:\.\d+)+)\.t)")
```

The trouble is that `page_match` is used in some strategies and not all of the strategies print debug output (e.g., "Possible SourceForge project..."), so seeing `page_match` in the output doesn't mean it's using the default strategy. To properly understand the strategy, you have to be familiar with the URL cases in `heuristic.rb` and think like the heuristic. This URL doesn't match any of the special cases (to select a specific strategy), so it uses the default `page_match` strategy on the URL.

This is a potential burden to people who aren't deeply invested in livecheck (e.g., someone adding a one-off livecheckable), so I would definitely like to improve the debug output in the future. I'll be doing some work on the heuristic for the `using` feature, so I'll either handle this at the same time or in a follow-up PR.

Reply:
Thanks! I almost wonder whether the `using` being mandatory would make the behaviour more obvious to non-livecheck experts.